analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/References.cs

```diff
@@ -66,37 +66,37 @@ protected override ProgramState Process(SymbolicContext context, IFieldReference
     }
 }
 
-internal sealed class PropertyReference : SimpleProcessor<IPropertyReferenceOperationWrapper>
+internal sealed class PropertyReference : BranchingProcessor<IPropertyReferenceOperationWrapper>
 {
     protected override IPropertyReferenceOperationWrapper Convert(IOperation operation) =>
         IPropertyReferenceOperationWrapper.FromOperation(operation);
 
-    protected override ProgramState Process(SymbolicContext context, IPropertyReferenceOperationWrapper propertyReference)
+    protected override ProgramState PreProcess(ProgramState state, IPropertyReferenceOperationWrapper operation)
     {
-        var state = context.State;
-        if (propertyReference.Instance.TrackedSymbol() is { } symbol)
+        if (operation.Instance.TrackedSymbol() is { } symbol)
         {
-            if (propertyReference.Instance.Type.IsNullableValueType())
+            if (!IsNullableHasValue(operation))
             {
-                if (propertyReference.Property.Name == "Value" && state[symbol] is { } value)
-                {
-                    state = state.SetOperationValue(context.Operation, value);
-                }
-                else if (propertyReference.Property.Name == "HasValue")
-                {
-                    // Return directly, do not set NotNull on the symbol itself
-                    return state[symbol]?.Constraint<ObjectConstraint>() is { } objectConstraint
-                        ? state.SetOperationConstraint(context.Operation, BoolConstraint.From(objectConstraint == ObjectConstraint.NotNull))
-                        : state;
-                }
+                state = state.SetSymbolConstraint(symbol, ObjectConstraint.NotNull);
+            }
+            if (operation.Property.Name == "Value" && state[symbol] is { } value)
```

Comment:
Misses the "IsNullableValueType()" check. For symmetry, consider introducing an `IsNullableValue(operation)` helper.

Reply:
Nice catch. As we need to check nullable & name, I've made the name a paramter